docs/dx.rst

```diff
@@ -0,0 +1,246 @@
+    DXCanonicalPath("dx://project-FJq288j0GPJbFJZX43fV5YP1:/")
+
+    >>> stor.Path('dx://project-FJq288j0GPJbFJZX43fV5YP1:/file-FKzjPgQ0FZ3VpBkpKJz4Vb70').virtual_path
+    DXVirtualPath("dx://StorTesting:/1.bam")
+
+
+The canonical_path and virtual_path attributes are cached and hence, each call to these
+properties doesn't invoke a new API request to the DX server.
+
+Directories on DNAnexus
+-----------------------
+DNAnexus has the concept of directories on the platform like posix (unlike
+Swift/S3). These directories can be empty, and also have names with extensions.
+Two folders with the same parent path cannot have the same name, i.e., no
+duplicate folders are allowed (like posix).
+
+Note here that folders are not actual resources on the DNAnexus platform.
+They are only handled through metadata on the server and as a result, do not
+have a canonical ID of their own. As a result, paths to folders can only be
+`DXVirtualPath`. Accessing the `DXVirtualPath.canonical_path` property for folders
+would raise an error.
+
+The summary of different behaviors for different filesystems is presented here. The
+individual details are explained further below.
+
+.. list-table:: Filesystem Differences
+   :widths: 15 30 25 30
+   :header-rows: 1
+
+   * - Topic
+     - Posix
+     - Swift/S3
+     - DNAnexus
+   * - File/ Directory names
+     - Filenames without extension and dirnames with ext are allowed.
+       A dir and a file can have the same name in same path
+     - Anonymous file / dir names are not allowed on swift paths
+     - Filenames without extension and dirnames with ext are allowed.
+       A dir and a file can have the same name in same path
+   * - Duplicates
+     - Not allowed
+     - Not allowed
+     - Duplicate filenames are allowed. Duplicate dir names are not allowed
+   * - list
+     - Recurisvely lists the files and empty directories
+     - Recurisvely lists the files and empty directory markers.
+     - Recursively lists the files within DX folder path. No empty
+       directories are listed.
+   * - list with prefix
+     - Treated as prefix to absolute path
+     - Treated as prefix to absolute path
+     - Treated as path to a subfolder to list
+   * - list on filepath
+     - Returns [filepath]
+     - Returns [filepath]
+     - Returns []
+   * - copy, target exists and is file
+     - Overwrites the file
+     - Overwrites the file
+     - Deletes the existing file before copying over
+   * - copy, target exists and is dir
+     - Not allowed
+     - Not possible
+     - Copied within existing dir
+   * - serverside copy/copytree
+     - Allowed
+     - Not allowed
+     - Allowed
+   * - copytree, target exists as file
+     - Not allowed
+     - Allowed
+     - Allowed
+   * - copytree, target exists as dir
+     - Merges the two directories
+     - Overwrites the dir
+     - Copies inside existing directory.
+       If root folder is moved, project name is used while copying if needed.
+
+Files on DNAnexus
```

Comment:
`Files on DNAnexus` section duplicates their docs. Let's move it to internal confluence and delete from here.

Reply:
Fixed.